tasks/setup-debian-vanilla.yml

```diff
@@ -10,12 +10,14 @@
     update_cache: yes
   tags:
     - wg-install
+  when: wireguard_debian_manage_deb_repo == "true"
```

Comment:
Please put https://github.com/githubixx/ansible-role-wireguard/blob/master/tasks/setup-debian-vanilla.yml#L6-L31 into a `block`. This avoids all that `when` clauses. The `(Debian) Get architecture` task is also only needed if the Linux headers are needed. An example of such a `block` is here: https://github.com/githubixx/ansible-role-wireguard/blob/master/tasks/setup-ubuntu.yml#L12-L24 in case you need the syntax.